dynamic_programming/knapsack.c

```diff
@@ -0,0 +1,227 @@
+/**
+ * @file
+ * @brief
+ * [0-1_Knapsack](https://en.wikipedia.org/wiki/Knapsack_problem#0-1_knapsack_problem) implementation
+ * @details
+ * From Wikipedia: The knapsack problem is
+ * Given a set of items, each with a weight and a value, determine
+ * which items to include in the collection
+ * so that the total weight is less than or equal to a given limit
+ * and the total value is as large as possible.
+ * @author [aa001R](https://github.com/aa001R)
+ */
+
+#include <assert.h>  /// for assert
+#include <stdio.h>   /// for IO operations
+
+/**
+ * @brief Returns the maximum value between two values a and b
+ * @param a the value to be compared
+ * @param b the value to be compared
+ * @returns the maximum value between a and b
+ */
+#define MAX(a, b) (((a) > (b)) ? (a) : (b))
+
+/**
+ * @brief Maximum capacity of the knapsack
+ */
+#define MAXCAPACITY 10
+
+/**
+ * @brief Item structure
+ */
+struct Item
+{
+    int weight;  ///< weight
+    int value;   ///< value
```

Comment:
You might want to initialize this at zero to prevent problems.
```suggestion
    int weight = 0;  ///< weight
    int value = 0;   ///< value
```

Reply:
Thank you for the suggestion😊!